zipkin-common/src/main/scala/com/twitter/zipkin/storage/SpanStore.scala

```diff
@@ -149,6 +149,23 @@ class InMemorySpanStore extends SpanStore with CollectAnnotationQueries {
       .toList
   }
 
+  override protected def getTraceIdsByDuration(
+    serviceName: String,
```

Comment:
If you have A->C->B, and any other trace involving B, I think the query by service B should always return all those traces if the B span was long. The owner of service B has no control over the shape of traces above B, so when they want to diagnose performance of B, they should see all traces containing B.